automation-tests/test/scratchOrg.ts

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import child_process from 'child_process';
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import {
+  DefaultTreeItem,
+  InputBox,
+  QuickOpenBox,
+  Workbench
+} from 'wdio-vscode-service';
+import {
+  EnvironmentSettings
+} from './environmentSettings';
+import {
+  utilities
+} from './utilities';
+
+const exec = util.promisify(child_process.exec);
+
+export class ScratchOrg {
+  private testSuiteSuffixName: string = undefined;
+  private reuseScratchOrg = false;
+  private projectFolderPath: string = undefined;
+  private prompt: QuickOpenBox | InputBox = undefined;
+  private scratchOrgAliasName: string = undefined;
+
+  public constructor(testSuiteSuffixName: string, reuseScratchOrg: boolean) {
+    this.testSuiteSuffixName = testSuiteSuffixName;
+    this.reuseScratchOrg = reuseScratchOrg;
+
+    // To have all scratch orgs be reused, uncomment the following line:
+    // this.reuseScratchOrg = true;
+  }
+
+  public async setUp(): Promise<void> {
+    await this.setUpTestingEnvironment();
+    await this.createProject();
+    await this.authorizeDevHub();
+    await this.createDefaultScratchOrg();
+  }
+
+  public async tearDown(): Promise<void> {
+    if (this.scratchOrgAliasName && !this.reuseScratchOrg) {
+      // To use VS Code's Terminal view, use:
+      // const workbench = await (await browser.getWorkbench()).wait();
+      // await utilities.executeCommand(workbench, `sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+
+      // The Terminal view can be a bit unreliable, so directly call exec() instead:
+      await exec(`sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+    }
+
+    if (this.projectFolderPath) {
+      await utilities.removeFolder(this.projectFolderPath);
+    }
+  }
+
+  public async setUpTestingEnvironment(): Promise<void> {
+    utilities.log('');
+    utilities.log(`${this.testSuiteSuffixName} - Starting setUpTestingEnvironment()...`);
+
+    const tempFolderPath = path.join(__dirname, '..', 'e2e-temp');
```

Comment:
In my project I found e2e-temp and e2etemp to be a bit confusing since they were so similar. Could we put it all under one folder with multiple projects? Or use more distinct names?

Reply:
seconded.  e2eProjects makes more sense to me.  Also if we commiting it to the repo temp is def confusing

Reply:
also should the value be a constant instead of a string value here?

Reply:
we could also use os.tmpdir() to get a temporary location that would be different between runs.